amida-tech/cql-execution
# Review 2nd: Blood Pressure Control HEDIS measure

# Related Tickets



- [SAR-75](https://jira.amida-tech.com/browse/SAR-75)

# Other Repos' PR(s) Intended to Work With This PR

# How Things Worked (or Didn't) Before This PR
There was no CQL and associated files for Diabetes BP Control

# How Things Work Now (And How to Test)
## How it works now
I've added the CQL and associated files for Diabetes BP Control

## To test
run `yarn` to install the necessary packages
run `node exec-files/exec-cdc_diabetes-bp.js`
Use your favorite diffchecker to confirm your output for patientResults and unfilteredResults matches what is below - you can ignore everything else.
```javascript
patientResults: {
    '96dcbd62-e4c6-a555-a663-77d08ad3c3b5': {
      Patient: [FHIRObject],
      'Initial Population': true,
      'Telehealth Modifier Encounter': true,
      'Telehealth POS Encounter': true,
      'Acute Inpatient Encounter': true,
      'Acute Inpatient Encounter Without Telehealth': false,
      'Acute Inpatient Discharge With Diabetes': [Array],
      'Claim with Acute Inpatient': [],
      'Acute Inpatient Discharge On Discharge Claim': false,
      'Nonacute Diabetes Discharges-List': [],
      'Nonacute Diabetes Discharges': 0,
      'Claim with Nonacute Inpatient Stay': [],
      'Nonacute Diabetes Discharges On Claim': 0,
      'Outpatient Visit': [Array],
      'Observation Visit': [],
      Out: [Array],
      'Telehealth Modifier With Diabetes': [Array],
      'Telephone Visit': [],
      'Online Assessment': [],
      'Out Tele Online with Different Dates': [],
      'Out Tele Online': 0,
      'At Least 2 Remote or Nonacute Inpatient Encounters With Diabetes': false,
      'Diabetes Diagnosis By Medication': true,
      Frailty: true,
      'Acute Inpatient With Advanced Illness': true,
      'Acute Inpatient Discharge With Advanced Illness': true,
      'Outpatient Advanced Illness': [Array],
      'Observation Advanced Illness': [],
      'ED Advanced Illness': [],
      'Nonacute AI': [],
      'Nonacute AI Discharge': [],
      'At Least Two Out/Obs/ED Advaned Illness': [],
      'Taking Dementia Medication': false,
      'Diabetes Exclusions': false,
      'Diabetes Condition': true,
      'Diabetes Condition Numerator': true,
      'Diabetes Condition Denominator': true,
      'Correct Context For BP Reading': true,
      'Qualifying Diastolic Blood Pressure Reading': [FHIRObject],
      'Qualifying Systolic Blood Pressure Reading': [FHIRObject],
      'BP < 140/90 mm Hg': true,
      Numerator: 1,
      Denominator: 1
    },
    '440487f4-1137-f542-8127-dc98e392d21e': {
      Patient: [FHIRObject],
      'Initial Population': true,
      'Telehealth Modifier Encounter': false,
      'Telehealth POS Encounter': false,
      'Acute Inpatient Encounter': true,
      'Acute Inpatient Encounter Without Telehealth': true,
      'Acute Inpatient Discharge With Diabetes': [Array],
      'Claim with Acute Inpatient': [],
      'Acute Inpatient Discharge On Discharge Claim': false,
      'Nonacute Diabetes Discharges-List': [Array],
      'Nonacute Diabetes Discharges': 1,
      'Claim with Nonacute Inpatient Stay': [],
      'Nonacute Diabetes Discharges On Claim': 0,
      'Outpatient Visit': [],
      'Observation Visit': [],
      Out: [],
      'Telehealth Modifier With Diabetes': [],
      'Telephone Visit': [],
      'Online Assessment': [],
      'Out Tele Online with Different Dates': [],
      'Out Tele Online': 0,
      'At Least 2 Remote or Nonacute Inpatient Encounters With Diabetes': false,
      'Diabetes Diagnosis By Medication': false,
      Frailty: true,
      'Acute Inpatient With Advanced Illness': false,
      'Acute Inpatient Discharge With Advanced Illness': false,
      'Outpatient Advanced Illness': [],
      'Observation Advanced Illness': [],
      'ED Advanced Illness': [],
      'Nonacute AI': [],
      'Nonacute AI Discharge': [],
      'At Least Two Out/Obs/ED Advaned Illness': [],
      'Taking Dementia Medication': false,
      'Diabetes Exclusions': false,
      'Diabetes Condition': true,
      'Diabetes Condition Numerator': true,
      'Diabetes Condition Denominator': false,
      'Correct Context For BP Reading': false,
      'Qualifying Diastolic Blood Pressure Reading': [FHIRObject],
      'Qualifying Systolic Blood Pressure Reading': [FHIRObject],
      'BP < 140/90 mm Hg': false,
      Numerator: 0,
      Denominator: 0
    }
  },
  unfilteredResults: {
    'Denominator Count': 1,
    'Numerator Count': 1,
    'CDC Measure Score': 1
  },
```
## To Review
Here's what to look for in each different file:

`exec-cdc_diabetes-bp.js` - This is the exec file that actually does the work - there's not much to review here.

`cdc_diabetes-bp.js` - The list of codes in each value set. This is programmatically generated using the CQL from the [VSAC](https://vsac.nlm.nih.gov/) API, so there's nothing to review here

`cdc_diabetes-bp.json` - JSON version of the CQL created using the [clinical_quality_language](https://github.com/cqframework/clinical_quality_language) - nothing to review here

`cdc_diabetes-bp-patients.js` - This contains the fake patients and their data. If you are really gung-ho, you can review the output of the results and confirm that the expressions with the CQL library returned the correct response based on the data, but that might take a while.

`cdc_diabetes-bp.cql` - this is the main piece of the PR. Read through the following requirements, and as best you can, confirm that the CQL matches the guidelines. If you'd prefer, whenever you get to this portion I can send you the word doc these requirements are in, which might be easier than looking at screenshots.
### NOTE - there are only a few small difference between this and the HbA1c CQL
Because of this, do a diff check between the hba1c cql file and the bp cql file and just review the bp requirements
<img width="691" alt="Screen Shot 2020-12-09 at 12 35 05 PM" src="https://user-images.githubusercontent.com/22867407/105083064-6bfd7680-5a62-11eb-9fad-543271a1e0d5.png">
